.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,55 @@
+name: Benchmark
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  id-token: write # required for OIDC
+  contents: read
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [3.2]
```

Comment:
We may as well run this on all 3.0+ based ruby versions

Reply:
Added